src/index/hnsw/faiss_hnsw.cc

```diff
@@ -1327,6 +1350,9 @@ class BaseFaissRegularIndexHNSWFlatNode : public BaseFaissRegularIndexHNSWNode {
             } else if (data_format == DataFormatEnum::bf16) {
                 hnsw_index = std::make_unique<faiss::IndexHNSWSQCosine>(dim, faiss::ScalarQuantizer::QT_bf16,
                                                                         hnsw_cfg.M.value());
+            } else if (data_format == DataFormatEnum::int8) {
+                hnsw_index = std::make_unique<faiss::IndexHNSWSQCosine>(
+                    dim, faiss::ScalarQuantizer::QT_8bit_direct_signed, hnsw_cfg.M.value());
```

Comment:
please **DO** confirm that you want to use `QT_8bit_direct_signed` here, because the use case is not clear to me. Basically, I can imagine a use case that works with the input data of `[0..255]` range (`QT_8bit_direct`), or the traditional `QT_8bit` that remaps input float values into `[0..255]` range, but what is the use case for the input data of `[-128..127]` range? Or is it just the requirement from Milvus?

Reply:
it's the requirement from Milvus, since vespa and qdrant already support Vector_Int8 now

Reply:
Hi Alex, I see no obvious difference between min() and lowest(), I prefer to use min() and max() in pair.